src/acrcssc/azext_acrcssc/helper/_workflow_status.py

```diff
@@ -0,0 +1,557 @@
+        try:
+            response = client.get_log_sas_url(
+                resource_group_name=resource_group_name,
+                registry_name=registry_name,
+                run_id=run_id)
+            log_file_sas = response.log_link
+        except (AttributeError, CloudError) as e:
+            logger.debug("%s Exception: %s", error_msg, e)
+            raise AzCLIError(error_msg)
+        except ResourceNotFoundError as e:
+            logger.debug("log file not found for run_id: %s, registry: %s, "
+                         "resource_group: %s -- exception: %s",
+                         run_id, registry_name, resource_group_name, e)
+
+        if await_task_run:
+            polling_method = WorkflowLogPollingMethod(client,
+                                                      resource_group_name,
+                                                      registry_name,
+                                                      run_id)
+            LongRunningOperation(
+                cmd.cli_ctx,
+                progress_bar=IndeterminateProgressBar(cmd.cli_ctx, message=await_task_message)
+            )(polling_method)
+
+        blobClient = get_sdk(cmd.cli_ctx, ResourceType.DATA_STORAGE_BLOB, '_blob_client#BlobClient')
+        return WorkflowTaskStatus._download_logs(blobClient.from_blob_url(log_file_sas))
+
+    @staticmethod
+    def evaluate_task_run_nonterminal_state(run_status):
+        return run_status != TaskRunStatus.Succeeded.value and run_status != TaskRunStatus.Failed.value
+
+    @staticmethod
+    def get_run_status_local(client, resource_group_name, registry_name, run_id):
+        try:
+            response = client.get(resource_group_name, registry_name, run_id)
+            return response.status
+        except (AttributeError, CloudError):
+            return None
+
+    @staticmethod
+    def _download_logs(blob_service):
+        blob = blob_service.download_blob()
+        blob_text = blob.readall().decode('utf-8')
+
+        return blob_text
+
+    @staticmethod
+    def remove_internal_acr_statements(blob_content):
+        lines = blob_content.split("\n")
+        starting_identifier = "DRY RUN mode enabled"
+        terminating_identifier = "Total matches found"
+        print_line = False
+        output = ""
+
+        for line in lines:
+            if line.startswith(starting_identifier):
+                print_line = True
+            elif line.startswith(terminating_identifier):
+                output += "\n" + line
+                print_line = False
+
+            if print_line:
+                output += "\n" + line
+
+        return output
+
+    @staticmethod
+    def _get_missing_taskrun(taskrun_client, registry, run_id):
+        try:
+            resourceid = parse_resource_id(registry.id)
+            resource_group = resourceid[RESOURCE_GROUP]
+            runs = WorkflowTaskStatus.get_taskruns_with_filter(taskrun_client,
```

Comment:
Is it possible runs is none or empty?

Reply:
yes, it was detected during testing.
Since the task runs are retrieved first and the log later, there is a window where a task run id is referenced in the logs but does not exist in the list of task runs we originally retrieved.
This function is to get the missing task runs